### ISSUE TITLE
Are 2FA secrets backed up with an encrypted Finder backup?

### ISSUE THREAD

**Issue by Karta775** [mattrubin/authenticator#374]
I don't know if/how iTunes and Finder backups differ but are the 2FA secrets **definitely** backed up in a Finder backup? I can no longer install iTunes on my machine so Finder is the only option and I don't have a spare or new iPhone to properly test my backup. Alternatively, will there ever be the option to export 2FA secrets raw or as QR codes to be scanned by a new device? It would be useful for switching phones or backing up the secrets. (edit: I don't know anything about iOS keychain so this could be a dumb suggestion)

Thanks :-)

**Comment by beaucollins** [comment]
TL;DR: yes


The only officially supported means of migrating your codes to a new device is through a secure backup. With iTunes no longer available Apple has moved this feature into Finder.

1. Plug in your phone to your computer
2. Open a Finder window
3. Choose your phone from 'Locations'
4. Enter any password prompts on either device to view the configuration screen

<img width="1056" alt="Pasted_Image_9_13_21__10_09_AM" src="https://user-images.githubusercontent.com/19795/133127609-dae0e484-cfe3-4ae5-a352-1d25ddd8e101.png" />

---

At some point over the past few years Apple has begun backing up these keychain items. https://github.com/mattrubin/Authenticator/issues/330#issuecomment-701124462

I use iCloud Keychain and have Apple's two factor system enabled on my iCloud account. I still make secure backups but through multiple device wipes and device upgrades I have never needed to use the encrypted backup. The codes are available if I use the iCloud account migration tools when setting up devices.

I suggest making secure backups still, but if you have iCloud keychain enabled along with Apple's two-factor authentication I believe you will find your Authenticator codes will be available when migrating to a new device.
